agent/stats/engine_unix_test.go

```diff
@@ -235,3 +247,176 @@ func TestFetchEBSVolumeMetrics(t *testing.T) {
 	assert.Len(t, actualMetrics, 1)
 	assert.Equal(t, actualMetrics, expectedMetrics)
 }
+
+// TestFetchEBSVolumeMetricsUnhappy tests that there was an issue with fetching volume metrics for an EBS volume via the CSI client. The expected behavior is
+// to have no volume metrics withn the output of fetchEBSVolumeMetrics.
+func TestFetchEBSVolumeMetricsUnhappy(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+	resolver := mock_resolver.NewMockContainerMetadataResolver(mockCtrl)
+	mockDockerClient := mock_dockerapi.NewMockDockerClient(mockCtrl)
+	t1 := &apitask.Task{
+		Arn: "t1",
+		Volumes: []apitask.TaskVolume{
+			{
+				Name: "1",
+				Type: apiresource.EBSTaskAttach,
+				Volume: &taskresourcevolume.EBSTaskVolumeConfig{
+					VolumeId:             "vol-12345",
+					VolumeName:           "test-volume",
+					VolumeSizeGib:        "10",
+					SourceVolumeHostPath: "taskarn_vol-12345",
+					DeviceName:           "/dev/nvme1n1",
+					FileSystem:           "ext4",
+				},
+			},
+		},
+	}
+
+	resolver.EXPECT().ResolveTaskByARN("t1").AnyTimes().Return(t1, nil)
+	mockDockerClient.EXPECT().Stats(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil).AnyTimes()
```

Comment:
Let's avoid using `AnyTimes()` unless necessary.

Reply:
(I'll also take that feedback on my own prs)

Reply:
done,this was really helpful in figuring out what exactly is needed and what is not 😄